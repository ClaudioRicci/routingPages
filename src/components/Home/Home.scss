@import "./../../variables.scss";

main {
  border: 1px solid $black;
  background: $skyblue;
  margin: 0 $medium;
  padding: $x-large;
}
