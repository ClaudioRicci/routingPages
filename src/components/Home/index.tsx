import React from "react";
import "./Home.scss";
import uuid from "uuid";
import { pure } from "recompose";

function Home() {
  return (
    <main key={uuid.v4()} role="main" data-testid="Home">
      <h1>Home Page</h1>
      <p>Please select a page from the tabs above</p>
    </main>
  );
}

export default pure(Home);
