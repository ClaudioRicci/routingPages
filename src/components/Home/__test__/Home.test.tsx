import React from "react";
import ReactDOM from "react-dom";
import Home from "..";
import { render, cleanup } from "@testing-library/react";
import "@testing-library/jest-dom/extend-expect";
import renderer from "react-test-renderer";

afterEach(cleanup);

it("Renders without crashing", () => {
  const div = document.createElement("div");
  ReactDOM.render(<Home></Home>, div);
});

test("Renders Home Correctly", done => {
  const { getByTestId } = render(<Home />);
  expect(getByTestId("Home")).toHaveTextContent("Home");
  done();
});

it("Matches snapshot", () => {
  const tree = renderer.create(<Home></Home>).toJSON();
  expect(tree).toMatchSnapshot();
});
