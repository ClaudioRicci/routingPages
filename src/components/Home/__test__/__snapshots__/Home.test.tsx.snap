// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Matches snapshot 1`] = `
<main
  data-testid="Home"
  role="main"
>
  <h1>
    Home Page
  </h1>
  <p>
    Please select a page from the tabs above
  </p>
</main>
`;
